tests/src/onelogin/models/app_rule_test.py

```diff
@@ -52,19 +47,19 @@ def testAppRuleAttributes(self):
 
 
     def testAppRulesV2Payload(self):
-        app_rule = AppRule(self.app_rules_v2_payload);
+        app_rule = AppRule(self.app_rules_v2_payload)
         expected_app_rule = self.getTestAppRulesV2()
         self.assertEqual(app_rule.actions,expected_app_rule.actions)
         app_rule.actions = expected_app_rule.actions = []
         self.assertEqual(app_rule.conditions,expected_app_rule.conditions)
         app_rule.conditions = expected_app_rule.conditions = []
-        self.assertEqual(unicode(app_rule), unicode(expected_app_rule))   
+        self.assertEqual(app_rule.__dict__, expected_app_rule.__dict__)   
```

Comment:
Can you give some background on the reason for changing it this way?

Reply:
Python dictionaries do not guarantee any particular order, so when you turn a dictionary into a string, that's an operation that isn't predictable. And so 2 different dictionaries which are equivalent don't always have the same string representation. Which causes these tests to fail *sometimes*. Comparing the dictionaries directly avoids this problem. You can actually see that it's been done to a few places already like here https://github.com/onelogin/onelogin-python-sdk/blob/v2/tests/src/onelogin/models/otp_device_test.py#L97 I've changed it to this behavior everywhere.